### PR TITLE
Update webpack-assets-manifest to webpack 5

### PR DESCRIPTION
**Pull request by simPod**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

### CODE REVIEW

**Comment by simPod** [comment]
_This also mirrors https://github.com/DefinitelyTyped/DefinitelyTyped/pull/52225_

**Comment by FranklinWhale** [comment]
I think this is going to be a [breaking change for webpack v3 and v4 users](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/52044#issuecomment-815401506), unless [this mechanism](DefinitelyTyped) is followed.

**Comment by FranklinWhale** [comment]
As [webpack-assets-manifest 5+ supports Webpack 5+ only](https://github.com/webdeveric/webpack-assets-manifest#new-in-version-5), I think a `v4` folder should be created.

**Comment by simPod** [comment]
@FranklinWhale created v4 folder

File: types/v4/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+    /** https://github.com/webdeveric/webpack-assets-manifest#hooks */
+    hooks: {
+        apply: SyncHook<WebpackAssetsManifest>;
+
+        /**
+         * The `SyncWaterfallHook` class supports 3 type parameters only but this hook actually has 4 parameters. The type of 4th parameter is `AnyObject`.
+         *
+         * Refer to https://github.com/webdeveric/webpack-assets-manifest#hooks for details
+         */
+        customize: SyncWaterfallHook<WebpackAssetsManifest.Entry, WebpackAssetsManifest.AnyObject, WebpackAssetsManifest>;
```
Comment (simPod) [inline types/v4/index.d.ts:22]: Any idea what arguments should be passed here? Webpack types allow 1-2, 3 are here though

File: types/webpack-assets-manifest/index.d.ts [hunk 2]
```diff
@@ -4,11 +4,13 @@
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 3.7
 
-import { Plugin } from "webpack";
+/// <reference types="node" />
+import { Compiler } from "webpack";
 import { SyncHook, SyncWaterfallHook } from "tapable";
 
-declare class WebpackAssetsManifest extends Plugin {
+declare class WebpackAssetsManifest {
```
Comment (simPod) [inline types/webpack-assets-manifest/index.d.ts:11]: Any idea what arguments should be passed here? Webpack types allow 1-2, 3 are here though

https://github.com/DefinitelyTyped/DefinitelyTyped/pull/52782/files#diff-0806b1c7645f696161960e912873b215e563758171ad7d3bdf414d3238daa1beR24

**Review by FranklinWhale — APPROVED** [review]
LGTM. Thanks!

**Comment by simPod** [comment]
Ready to merge

**Pull request merged**